lib/solveEquation/stepThrough.js

```diff
@@ -72,31 +71,45 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
 
     try {
       // Find roots of a equation
-      // Conditions: left side is a mul node, right side is 0
+      // Conditions: left side is a multiplication node, right side is 0
       // TODO: move polynomial to left if is on the right
       // TODO: assuming a single symbol in equation
       // TODO: display all solutions
+      // TODO: add optional param for Node.Type.isConstant
       // Returns: An equation in the form of symbol = root
       if (equation.leftNode.op === '*'
-          && (Node.Type.isConstant(equation.rightNode) && parseFloat(equation.rightNode.value) === 0)) {
+          && (Node.Type.isConstant(equation.rightNode)
+              && parseFloat(equation.rightNode.value) === 0)) {
         // Gets all factors that aren't a constant
         const factors = equation.leftNode.args.filter(arg => !Node.Type.isConstant(arg));
 
         const solutions = [];
-
         let symbol;
+
         // For each factor, solve the equation factor = 0 for the symbol
         for (var f in factors) {
-          const leftNode = factors[f].content;
+          const leftNode = Node.Type.isParenthesis(factors[f])
+                ? factors[f].content
+                : factors[f];
+
           const rightNode = Node.Creator.constant(0);
 
-          const lastStep = stepThrough(leftNode, rightNode, '=').slice(-1)[0];
+          const allSteps = stepThrough(leftNode, rightNode, '=');
+
+          if (allSteps.length > 0) {
+            const lastStep = stepThrough(leftNode, rightNode, '=').slice(-1)[0];
 
-          symbol = lastStep.newEquation.leftNode;
-          // Append to a list of roots
-          solutions.push(lastStep.newEquation.rightNode);
+            symbol = lastStep.newEquation.leftNode;
+            // Append to a list of roots
+            solutions.push(lastStep.newEquation.rightNode);
+          }
+          else {
+            // e.g. x (x - 2) = 0, x = 0
```

Comment:
why is this the case for whenever the else catches? what if it was 2^x = 0 --> there'd be no steps, but 0 isn't a root